napari/layers/base/base.py

```diff
@@ -905,15 +912,51 @@ def help(self, help_text):
 
     @property
     def interactive(self):
-        """bool: Determine if canvas pan/zoom interactivity is enabled."""
-        return self._interactive
+        warnings.warn(
+            '`Layer.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0.',
+            category=DeprecationWarning,
+        )
+        return self.mouse_pan or self.mouse_zoom
 
     @interactive.setter
     def interactive(self, interactive):
-        if interactive == self._interactive:
+        warnings.warn(
+            '`Layer.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0.',
+            category=DeprecationWarning,
+        )
+        self.mouse_pan = interactive
+        self.mouse_zoom = interactive
+
+    @property
+    def mouse_pan(self):
```

Comment:
```suggestion
    def mouse_pan(self) -> bool:
```

Reply:
The same situation as with VisPyCamera, it doesn't match the style of the rest of code. However, in all fairness, this file is already mixing type annotations here and there.